geomesa-spark/geomesa-spark-jts/src/main/scala/org/locationtech/geomesa/spark/jts/rules/GeometryLiteralRules.scala

```diff
@@ -22,20 +24,30 @@ object GeometryLiteralRules {
     override def apply(plan: LogicalPlan): LogicalPlan = {
       plan.transform {
         case q: LogicalPlan => q.transformExpressionsDown {
-          case s: ScalaUDF =>
-            // TODO: Break down by GeometryType
-            Try {
-                s.eval(null) match {
-                  case row: GenericInternalRow =>
-                    val ret = GeometryUDT.deserialize(row)
-                    GeometryLiteral(row, ret)
-                  case other: Any =>
```

Comment:
@elahrvivaz Hmm....  do we have an UDFs where the s.dataType is not a JTS Geometry?  

I'm trying to figure out if this 'fall back' case is still being applied after the refactoring.

Reply:
i believe that the 'if' guards in the case statements will handle that. since transformExpressionsDown takes a partial function, it will just skip over non-geometry data types. since previously we matched on any scalaUDF, we had to handle the non-geometry types as well

Reply:
but to answer your question, yes there are scala udfs with literals in the unit tests that started failing before i fixed this block.